### PR TITLE
Cap 68  cursor survival

### PR DESCRIPTION
**Pull request by smitchell1980**
Added reconnectDB, check_conn, sql_err functions.  They can be called if the connection is closed unexpectedly.  Modified all functions that try to access the DB with checks to make sure the connection is alive, if dead then tries to reconnect. if can't reconnect after 30 attempts send failure message back to web server to alert user.  all queries have exceptions to rollback if bad sql command given.

### CODE REVIEW

File: driver.py [hunk 1]
```diff
@@ -25,6 +25,59 @@
...
+
+def sql_except(err):
+    """
+    Function to print out the error message generated from the exception
+    Args:
+        err - the error message generated
+    Returns None
+    """
+    # get the details for exception
+    err_type, err_obj, traceback=sys.exc_info()
```
Comment (anhaley) [inline driver.py:74]: Are we doing anything with this info?

Comment (smitchell1980) [reply driver.py]: I was using it for making sure the exception was being caught on my ide console.  We can send it back to the server to display to user if you want.

**Comment by smitchell1980** [comment]
So I just tried to build the branch and test and I keep getting errors on the POST endpoint.  Not sure why so going to hold on this pull request until I figure out what is going wrong.

**Comment by smitchell1980** [comment]
Ok so looks like at somepoint a second instance of the insert command for the metadata table was added into the function call.  But that was causing an issue because it didn't have the new rows/columns attr in the command.  Should be fixed now and ready to merge once again.

**Pull request merged**